src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -51,10 +53,19 @@
         minor (.getDatabaseMinorVersion db-metadata)]
     [major minor]))
 
+(defn get-db-metadata
+  []
+  (let [db-metadata (.. (sql/find-connection)
```

Comment:
based on messing with this for a while today, I think it'll be cleaner to change this function to accept an actual connection object that we'll have once we shift from the undeprecated with-db-connection (which uses a binding) instead of with-connection like we currently do.  We'll need to make a similar change in a lot of places, and I haven't found any mixed approaches that I like better than what's currently here.